File: 이건희/level2/할인 행사.py
from collections import Counter

def solution(want, number, discount):
    result = 0
    
    for i in range(len(discount) - 9):
        things = Counter(discount[i : i + 10])
        cnt = 0
        for i in range(len(want)):
            if things[want[i]] == number[i]:
                cnt += 1
        if cnt == len(want):
            result += 1
            
    return result

File: 이훈일/Programmers/Level2/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        HashMap<String, Integer> wants = new HashMap<>();
        for (int i = 0; i < want.length; i++) {
            wants.put(want[i], number[i]);
        }
        
        int result = 0;
        for (int i = 0; i <= discount.length - 10; i++) {
            HashMap<String, Integer> discounts = new HashMap<>();
            for (int j = 0; j < 10; j++) {
                String goods = discount[i + j];
                if (discounts.containsKey(goods)) {
                    discounts.put(goods, discounts.get(goods) + 1);
                } else {
                    discounts.put(goods, 1);
                }
                    
            }
            
            boolean isPossible = true;
            for (String goods : wants.keySet()) {
                if (wants.get(goods) != discounts.get(goods)) {
                    isPossible = false;
                    break;
                }
            }
            
            if (isPossible) {
               result++; 
            }
        }
        
        return result;
    }
}
